### PR TITLE
Fix some CMake warnings related to set() + docstrings

### PR DESCRIPTION
**Pull request by Simran-B**
### Scope & Purpose

Fix some warnings in CMakeLists.txt by correcting the CACHE types and adding some doc strings - feel free to improve them!

Removes CMake warnings like this one:

```
CMake Warning (dev) at 3rdParty/s2geometry/CMakeLists.txt:11 (set):
  implicitly converting 'type' to 'STRING' type.
This warning is for project developers.  Use -Wno-dev to suppress it.
```

Also fixed typos and improved the program friendly strings (Windows only).

- [x] Bug-Fix for *devel-branch*
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)

### Testing & Verification

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

**Comment by Simran-B** [comment]
I used `cmake version 3.14.5` on Windows 10 with MSVC / VS Community Edition 16 2019.

There are additional warnings, but I don't know how to fix them:

```
CMake Warning (dev) at 3rdParty/iresearch/CMakeLists.txt:91 (find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  CMake variable BFD_ROOT is set to:

    invalid

  For compatibility, CMake is ignoring the variable.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Could NOT find BFD (missing:  BFD_INCLUDE_DIR BFD_SHARED_LIBS BFD_STATIC_LIBS BFD_INCLUDE_DIR_ANSIDECL BFD_INCLUDE_DIR_BFD BFD_SHARED_LIB BFD_STATIC_LIB BFD_STATIC_LIB_IBERTY BFD_STATIC_LIB_Z)
BFD_INCLUDE_DIR:
BFD_LIBRARY_DIR:
BFD_SHARED_LIBS:
BFD_STATIC_LIBS:
BFD_SHARED_LIB_RESOURCES:
```

```
CMake Warning (dev) at 3rdParty/lz4/contrib/cmake_unofficial/CMakeLists.txt:97 (option):
  Policy CMP0077 is not set: option() honors normal variables.  Run "cmake
  --help-policy CMP0077" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  For compatibility with older versions of CMake, option is clearing the
  normal variable 'LZ4_POSITION_INDEPENDENT_LIB'.
```

```
CMake Warning (dev) at 3rdParty/iresearch/CMakeLists.txt:110 (find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  CMake variable ICU_ROOT is set to:

    C:/Daten/ArangoDB/arangodb/3rdParty/V8/v7.1.302.28/third_party/icu/

  For compatibility, CMake is ignoring the variable.
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -296,18 +296,18 @@ set(ARANGODB_PACKAGE_VENDOR  "ArangoDB GmbH")
...
 set(ARANGODB_CONTACT         "hackers@arangodb.com")
 set(ARANGODB_FRIENDLY_STRING "ArangoDB - the native multi-model NoSQL database")
 
 # MSVC
 set(ARANGO_BENCH_FRIENDLY_STRING   "arangobench - stress test program")
-set(ARANGO_DUMP_FRIENDLY_STRING    "arangodump - export")
-set(ARANGO_RESTORE_FRIENDLY_STRING "arangrestore - importer")
-set(ARANGO_EXPORT_FRIENDLY_STRING  "arangoexport - datae xporter")
-set(ARANGO_IMPORT_FRIENDLY_STRING  "arangoimport - TSV/CSV/JSON importer")
+set(ARANGO_DUMP_FRIENDLY_STRING    "arangodump - backup creation tool")
```
Comment (KVS85) [inline CMakeLists.txt]: `data dump creation tool` to  leave space for `arangobackup`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -296,18 +296,18 @@ set(ARANGODB_PACKAGE_VENDOR  "ArangoDB GmbH")
...
+set(ARANGO_RESTORE_FRIENDLY_STRING "arangrestore - backup recovery tool")
```
Comment (KVS85) [inline CMakeLists.txt]: `data dump restoration tool` to  leave space for `arangobackup`.

File: CMakeLists.txt (hunk 1 above)
Comment (Simran-B) [reply CMakeLists.txt]: It does not only dump data, also collection structure etc. Can we call it "database dumping tool"?

**Comment by Simran-B** [comment]
Remaining warnings:

```
CMake Warning (dev) at 3rdParty/iresearch/CMakeLists.txt:91 (find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  CMake variable BFD_ROOT is set to:

    invalid

  For compatibility, CMake is ignoring the variable.
This warning is for project developers.  Use -Wno-dev to suppress it.

BFD_INCLUDE_DIR:
BFD_LIBRARY_DIR:
BFD_SHARED_LIBS:
BFD_STATIC_LIBS:
BFD_SHARED_LIB_RESOURCES:
-- Could NOT find BFD (missing:  BFD_INCLUDE_DIR BFD_SHARED_LIBS BFD_STATIC_LIBS BFD_INCLUDE_DIR_ANSIDECL BFD_INCLUDE_DIR_BFD BFD_SHARED_LIB BFD_STATIC_LIB BFD_STATIC_LIB_IBERTY BFD_STATIC_LIB_Z)
CMake Warning (dev) at 3rdParty/lz4/contrib/cmake_unofficial/CMakeLists.txt:97 (option):
  Policy CMP0077 is not set: option() honors normal variables.  Run "cmake
  --help-policy CMP0077" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  For compatibility with older versions of CMake, option is clearing the
  normal variable 'LZ4_POSITION_INDEPENDENT_LIB'.
This warning is for project developers.  Use -Wno-dev to suppress it.
```

```
CMake Warning (dev) at 3rdParty/iresearch/CMakeLists.txt:110 (find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  CMake variable ICU_ROOT is set to:

    /work/ArangoDB/3rdParty/V8/v7.1.302.28/third_party/icu/

  For compatibility, CMake is ignoring the variable.
This warning is for project developers.  Use -Wno-dev to suppress it.
```

**Pull request merged**